krm-functions/lib/interface/v1alpha1/interface.go

```diff
@@ -182,28 +176,23 @@ func (r *itfce) SetSpec(spec *nephioreqv1alpha1.InterfaceSpec) error {
 		if err := r.SetNetworkInstanceName(string(spec.NetworkInstance.Name)); err != nil {
 			return err
 		}
-	} else {
-		return fmt.Errorf("networkInstance is required")
 	}
 	return nil
 }
 
 // DeleteAttachmentType deletes the attachmentType from the spec
 func (r *itfce) DeleteAttachmentType() error {
-	return r.deleteNestedField(attachmentType...)
+	return r.deleteNestedField(cniType...)
 }
 
-// DeleteAttachmentType deletes the attachmentType from the spec
+// DeleteCNIType deletes the cniType from the spec
 func (r *itfce) DeleteCNIType() error {
-	return r.deleteNestedField(cniType...)
+	return r.deleteNestedField(attachmentType...)
```

Comment:
Looks like a typo here. "cniType" needs to be passed in.

Reply:
fixed + added test to cover this